src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -9937,3 +9938,118 @@ void ObjectMgr::LoadSceneTemplates()
+
+    if (!choices)
+    {
+        TC_LOG_INFO("server.loading", ">> Loaded 0 player choices. DB table `playerchoice` is empty.");
+        return;
+    }
+
+    uint32 responseCount = 0;
+    uint32 rewardCount = 0;
+
+    do
+    {
+        Field* fields = choices->Fetch();
+
+        int32 choiceId = fields[0].GetInt32();
+
+        PlayerChoice& choice = _playerChoiceContainer[choiceId];
+        choice.ChoiceId = choiceId;
+        choice.Question = fields[1].GetString();
+    } while (choices->NextRow());
+
+    TC_LOG_INFO("server.loading", ">> Loaded " SZFMTD " player choices in %u ms.", _playerChoiceContainer.size(), GetMSTimeDiffToNow(oldMSTime));
+
+    QueryResult responses = WorldDatabase.Query("SELECT ChoiceID, ResponseID, ChoiceArtFileID, Header, Answer, Description, Confirmation FROM playerchoice_response");
+    if (responses)
+    {
+        do
+        {
+            ++responseCount;
+            Field* fields = responses->Fetch();
+
+            int32 choiceId      = fields[0].GetInt32();
+            int32 ResponseID    = fields[1].GetInt32();
+
+            if (_playerChoiceContainer.find(choiceId) == _playerChoiceContainer.end())
+            {
+                TC_LOG_ERROR("sql.sql", "Table `playerchoice_response` has nonexistent choiceId: %u, skipped", choiceId);
+                continue;
+            }
+
+            PlayerChoice& choice = _playerChoiceContainer[choiceId];
+            PlayerChoiceResponse& response = choice.Responses[ResponseID];
+
+            response.ResponseID         = ResponseID;
+            response.ChoiceArtFileID    = fields[2].GetInt32();
+            response.Header             = fields[3].GetString();
+            response.Answer             = fields[4].GetString();
+            response.Description        = fields[5].GetString();
+            response.Confirmation       = fields[6].GetString();
+        } while (responses->NextRow());
+
+        TC_LOG_INFO("server.loading", ">> Loaded %u player choices response in %u ms.", responseCount, GetMSTimeDiffToNow(oldMSTime));
+    }
+    else
+    {
+        TC_LOG_INFO("server.loading", ">> Loaded 0 player choices responses. DB table `playerchoice_response` is empty.");
+    }
+
+    QueryResult rewards = WorldDatabase.Query("SELECT ChoiceID, ResponseID, TitleID, PackageID, SkillLineID, SkillPointCount, ArenaPointCount, HonorPointCount, Money, Xp FROM playerchoice_response_reward");
+
+    if (rewards)
+    {
+        do
+        {
+            ++rewardCount;
+            Field* fields = rewards->Fetch();
+
+            int32 choiceId      = fields[0].GetInt32();
+            int32 ResponseID    = fields[1].GetInt32();
+
+            if (_playerChoiceContainer.find(choiceId) == _playerChoiceContainer.end())
+            {
+                TC_LOG_ERROR("sql.sql", "Table `playerchoice_response_reward` has nonexistent choiceId: %u, skipped", choiceId);
```

Comment:
Always log the entire PK in case of error to be able to quickly identify offending rows